src/ft2/_ft2.cpp

```diff
@@ -0,0 +1,635 @@
+#include "_ft2.h"
+#include "_layout.h"
+
+namespace matplotlib::ft2 {
+
+Face::Face(std::string const& path, double hinting_factor) :
+  ptr{
```

Comment:
I do not understand why `shared_ptr` is needed if you are initializing it in the object constructor...

Reply:
Because it's a bit more convenient if a copy constructor is available (also, pybind11 requires either the move or the copy ctor to exist).  So you either need to use a raw pointer and call FT_Reference_Face in the copy ctor and then FT_Done_Face on the way out, or use a shared_ptr+custom deleter and the default copy/move ctors which do the job for you.